.github/workflows/release-pr.yml

```diff
@@ -1,88 +1,107 @@
+        default: 0.0.0
+        type: string
+
+  workflow_call:
+    inputs:
+      extension_version:
+        default: 0.0.0
+        type: string
+
+      sdk_version:
+        default: 0.0.0
+        type: string
 
 jobs:
-    release-pr:
-        runs-on: "macos-latest"
-        strategy:
-            matrix:
-                node-version: [16.x]
-        steps:
-            - uses: actions/checkout@v3
-              with:
-                  fetch-depth: 0
-
-            - run: git fetch --all --tags
-
-            - name: Check Version
-              run: bash -c '$([[ "${{ inputs.version }}" =~ ^([0-9]+\.){2}[0-9]+ ]])'
-
-            - run: |
-                  git config --global user.name "releasebot"
-                  git config --global user.email "noreply@github.com"
-
-            - name: Use Node.js ${{ matrix.node-version }}
-              uses: actions/setup-node@v3
-              with:
-                  node-version: ${{ matrix.node-version }}
-                  cache: "yarn"
-
-            - run: yarn install --immutable
-
-            - name: Bump version
-              if: ${{ inputs.version != '0.0.0' }}
-              run: |
-                  yarn workspaces foreach version ${{ inputs.version }}
-                  yarn version ${{ inputs.version }}
-
-            - name: Changelog
-              id: create-changelog
-              run: |
-                  TMPFILE=$(mktemp /tmp/tmpfile.XXXX)
-                  bash scripts/generate_changelog.sh ${{ inputs.version }} $TMPFILE
-
-                  cat $TMPFILE >> $GITHUB_STEP_SUMMARY
-
-                  echo "changelog_file=$TMPFILE" >> $GITHUB_OUTPUT
-
-            - name: Generate notice file
-              run: yarn run generate-notice
-
-            - name: lint fix
-              if: ${{ inputs.version != '0.0.0' }}
-              run: yarn run fix
-
-            - name: Create Branch
-              if: ${{ inputs.version != '0.0.0' }}
-              id: create-branch
-              run: |
-                  COMMIT_SHA=$(git rev-parse --short HEAD)
-                  BRANCH=releases/draft-$COMMIT_SHA-$(date +%s)
-                  git checkout -b $BRANCH
-                  git add  **/CHANGELOG.md **/package.json package.json **/NOTICE.md
-                  git status
-                  git commit -m "Changelog & version bump to ${{ inputs.version }}"
-                  git push origin HEAD
-                  echo "branch_name=$BRANCH" >> $GITHUB_OUTPUT
-
-            - name: Create PR
-              if: ${{ inputs.version != '0.0.0' }}
-              run: |
-                  gh pr create -B main -H ${{ steps.create-branch.outputs.branch_name }} --title "Release: v${{ inputs.version }}" --body-file ${{ steps.create-changelog.outputs.changelog_file }}
-                  rm ${{ steps.create-changelog.outputs.changelog_file }}
-              env:
-                  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  release-pr:
+    runs-on: "macos-latest"
```

Comment:
nit: can we make it run on Linux? Faster and cheaper.

Reply:
Not yet. Linux has slightly different behaviour for shell commands (eg awk), so it is hard to test. We can do it in a followup when I am more confident with the devloop.